### PR TITLE
#13: parse mentions (closes #13)

### LINKED ISSUE #13: parse mentions
## requirements
transform mentions

from:
```md
...
my friend @giampierone told me that elephants can fly
...
```
to:
```md
...
my friend [@giampierone](https://medium.com/@giampierone) told me that elephants can fly
...
```

## specs
{optional: describe technical specs to implement this feature, if not obvious}

## misc
{optional: other useful info}

### PR DESCRIPTION
**Pull request by francescogior**
Closes #13

## Test Plan

> These results can be screenshots, query results, or even just pointers to unit tests that successfully passed.}

> An example of partial test is a field displaying 4 possible values. If 3 values are tested, with screenshots, and 1 is not, then it should be mentioned here.}

### CODE REVIEW

File: src/parseAndModify/parseMentions.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+module.exports = (articleString) => {
+  const inlineCodes = [];
+  const blockCodes = [];
+  return articleString
+    .replace(inlineCodeRe, (_, code) => { inlineCodes.push(code); return inlineCodePlaceholder; })
+    .replace(blockCodeRe, (_, code) => { blockCodes.push(code); return blockCodePlaceholder; })
+    .replace(mentionRe, (_, s, username) => `${s}[@${username}](https://medium.com/@${username})` )
+    .replace(inlineCodePlaceholderRe, inlineCodes.shift())
+    .replace(blockCodePlaceholderRe, blockCodes.shift());
```
Comment (giogonzo) [inline src/parseAndModify/parseMentions.js]: looks like you're replacing every inlineCode and every blockCode with the first item of inlineCodes and blockCodes respectively

Comment (francescogior) [reply src/parseAndModify/parseMentions.js]: Why? `Array.prototype.shift` returns the first element of the array and removes it.

Comment (giogonzo) [reply src/parseAndModify/parseMentions.js]: but you're calling shift a single time

Comment (francescogior) [reply src/parseAndModify/parseMentions.js]: You're right 🤔

**Review by giogonzo — APPROVED** [review]
looks better

https://xkcd.com/208/

**Pull request merged**